src/Adapter/Order/CommandHandler/IssuePartialRefundHandler.php

```diff
@@ -112,15 +115,38 @@ public function handle(IssuePartialRefundCommand $command)
         }
 
         // Create order slip
-        $this->orderSlipCreator->createOrderSlip($order, $orderRefundDetail);
+        if ($command->generateCreditSlip()) {
+            $this->orderSlipCreator->create($order, $orderRefundSummary);
+        }
+
+        // Update order details (after credit slip to avoid updating refunded quantities while the credit slip fails)
+        foreach ($orderRefundSummary->getProductRefunds() as $orderDetailId => $productRefund) {
+            $orderDetail = $orderRefundSummary->getOrderDetailById($orderDetailId);
+            if ($order->hasBeenPaid()) {
```

Comment:
Prevent useless indentation
```suggestion
            if (!$order->hasBeenPaid()) {
                continue;
            }
```

Reply:
Actually moved it in a method so it's cleaner